### PR TITLE
Fixing Z level cable update and substations

### LINKED ISSUE #5165: Power networks not seperated by breakers
For ages (months to a year) now, the powernetworks does not seem to work as they should.

What should happen:
- Disabling the breaker should seperate the network behind the breaker, from the main grid (or the network before the breaker to be more precise). This is the case on upstream (bay) and on most of the breakers currently there.
- This should mean that the power network monitor should only show the APC's on the seperated network behind the breaker

What does happen:
- Some of the networks tend to stay connected with APC's outside their powernetwork, even when the breakers is disabled. 
- More interesting it seems that even when all breakers are disabled, some breakers tend to connect to APC's on other powernetworks that should be blocked by two disabled breakers!?

What research has been done:
- I checked the map for wires interconnecting said networks, and there are none.
- tested it ingame, time and time again.

Above behaviour is a shame because it limits the ability of engineers, antags and AI player to influence the powerflow in a more creative maner.

### LINKED ISSUE #5655: Main Science/Civilian Surface Powergrid does not work
The Main Science and Civilian Surface powergrids appear to be wired incorrect. When set up, they still try to send power to the Master powergrid, which in turn tries to power those two powergrids, which causes an infinite loop. This doesn't cause any damage but it basicly means that those two grids can never be set up. 

My best guess on this is that both grids have a cable somewhere that connects back to the master grid instead of going through the substations. I have no idea where it could be however as I've searched IC for the problem a few times now.

### LINKED ISSUE #5883: Qdel of power cable does not update powernet propely through Z level
When you cut/delete/remove cable, it does not properly update the powernet. Making one side of the net still connected to the other side, but the other side looses a chunk of net that creates a third it's own net that is not connected to anything.

Steps to reproduce:
Best example is to cut cable on the T-comms level that connects main grid to the T-comms. The T-comms side remains connected, while 10 cables that go to maingrid will create their own net that is not connected code-wise to the main grid(physically is).

### PR DESCRIPTION
**Pull request by Sindorman**
Oh boi, this one is a tiny in terms of lines, yet a huge impact. It absolutely solves problem with powernet bug of cables not updating through Z levels properly. Spend lots of time and especially intellectual effort to debug this thing and finally comparing Bay's code and debugging ours showed that lines 359 and 364 should be `power_list(T, src, 12)` because that stands for marked and we have unmarked cables.

- Fixes Z level update of cables, now it updates them properly. So the substations work. Fixes #5165 , Fixes #5883 , Fixes #5655 

- Fixes mapping of cables

- Increases charge of runtime SMES

- Adds Z powernet to runtime levels for debug and tests

- Adds Z level supply pipes and vents for debugging and tests

TO-DO:

- [x] Check if substations work

- [x] Check if cutting/qdel/explosion of cables update Z level networks

### CODE REVIEW

**Comment by skull132** [comment]
That's an interesting test error.

File: code/modules/power/cable.dm [hunk 1]
```diff
@@ -356,12 +356,12 @@ obj/structure/cable/proc/cableColor(var/colorC)
 	if(d1 == 11 || d2 == 11)
 		T = GetBelow(src)
 		if(T)
-			. += power_list(T, src, 12, 1)
+			. += power_list(T, src, 12)
```
[the remark this answers is not in the document]
Comment (skull132) [reply code/modules/power/cable.dm]: I mean, if there's a parameter for it, then why _not_ pass it down? Specially if it has a sane default value already.

[the remark this answers is not in the document]
Comment (Sindorman) [reply code/modules/power/cable.dm]: It technically doesn't matter if we put 0 there as the default is 0 anyway.

[the remark this answers is not in the document]
Comment (Lohikar) [reply code/modules/power/cable.dm]: ...that's not at all what was being suggested.

File: code/modules/power/cable.dm [hunk 2]
```diff
@@ -356,12 +356,12 @@ obj/structure/cable/proc/cableColor(var/colorC)
 	if(d1 == 11 || d2 == 11)
 		T = GetBelow(src)
 		if(T)
-			. += power_list(T, src, 12, 1)
+			. += power_list(T, src, 12, 0)
```
Comment (Lohikar) [inline code/modules/power/cable.dm]:
```suggestion
			. += power_list(T, src, 12, powernetless_only)
```

File: code/modules/power/cable.dm [hunk 3]
```diff
@@ -356,12 +356,12 @@ obj/structure/cable/proc/cableColor(var/colorC)
...
 		T = GetBelow(src)
 		if(T)
-			. += power_list(T, src, 12, 1)
+			. += power_list(T, src, 12, 0)
 
 	if(d1 == 12 || d2 == 12)
 		T = GetAbove(src)
 		if(T)
-			. += power_list(T, src, 11, 1)
+			. += power_list(T, src, 11, 0)
```
Comment (Lohikar) [inline code/modules/power/cable.dm]:
```suggestion
			. += power_list(T, src, 11, powernetless_only)
```

File: code/modules/power/cable.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sindorman) [reply code/modules/power/cable.dm]: Oh I see, I am sorry. I understood it wrong of what you wanted me to do. Let me test your suggestion

[the remark this answers is not in the document]
Comment (Sindorman) [reply code/modules/power/cable.dm]: You are right, I have tested your suggestion and it works as intended without issue. Pushed

**Pull request merged**